src/main/java/enmasse/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -293,22 +293,26 @@ public void handle(ProtonLink<?> link) {
 								Object obj = this.vertx.sharedData().getLocalMap(this.context.getEbName()).remove(deliveryTag);
 								
 								if (obj instanceof KafkaMessage<?, ?>) {
-	
+									// get partition and offset from delivery tag : <partition>_<offset>
+									int underscore = deliveryTag.indexOf("_");
+									int messagePartition = Integer.valueOf(deliveryTag.substring(0, underscore));
+									long messageOffset = Long.valueOf(deliveryTag.substring(underscore + 1));
```

Comment:
yes ... sorry you are right. Sometime we should check if we can move this "big" inline callback to a method in order to avoid something like that, because offset and partition are declared outside the callback here. For now leave as it is please.